src/OrleansAWSUtils/Storage/DynamoDBDataManager.cs

```diff
@@ -0,0 +1,248 @@
+
+        private async Task<TableDescription> GetTableDescription()
+        {
+            try
+            {
+                var description = await ddbClient.DescribeTableAsync(TableName);
+                if (description.Table != null)
+                    return description.Table;
+            }
+            catch (ResourceNotFoundException)
+            {
+                return null;
+            }
+            return null;
+        }
+
+        private async Task CreateTable()
+        {
+            var request = new CreateTableRequest
+            {
+                TableName = TableName,
+                AttributeDefinitions = new List<AttributeDefinition>()
+                {
+                    new AttributeDefinition
+                    {
+                        AttributeName = "PartitionKey",
+                        AttributeType = "S"
+                    },
+                    new AttributeDefinition
+                    {
+                        AttributeName = "RowKey",
+                        AttributeType = "S"
+                    }
+                },
+                KeySchema = new List<KeySchemaElement>()
+                {
+                    new KeySchemaElement
+                    {
+                        AttributeName = "PartitionKey",
+                        KeyType = "HASH"
+                    },
+                    new KeySchemaElement
+                    {
+                        AttributeName = "RowKey",
+                        KeyType = "RANGE"
+                    }
+                },
+                ProvisionedThroughput = new ProvisionedThroughput
+                {
+                    ReadCapacityUnits = readCapacityUnits,
+                    WriteCapacityUnits = writeCapacityUnits
+                }
+            };
+
+            try
+            {
+                var response = await ddbClient.CreateTableAsync(request);
+                TableDescription description = null;
+                do
+                {
+                    description = await GetTableDescription();
+
+                    await Task.Delay(2000);
+
+                } while (description.TableStatus != TableStatus.ACTIVE);
+            }
+            catch (Exception exc)
+            {
+                Logger.Error(ErrorCode.StorageProviderBase, $"Could not create table {TableName}", exc);
+                throw;
+            }
+        }
+
+        public Task DeleteTableAsync()
+        {
+            return ddbClient.DeleteTableAsync(new DeleteTableRequest { TableName = TableName });
+        }
+
+        public Task UpsertItemsAsync(IEnumerable<T> entries)
+        {
+            var config = new DynamoDBOperationConfig { ConsistentRead = true, SkipVersionCheck = true, OverrideTableName = TableName };
+            var batch = ddbContext.CreateBatchWrite<T>(config);
+            batch.AddPutItems(entries);
+            return batch.ExecuteAsync();
+        }
+
+        public Task UpsertItemAsync(T entry)
```

Comment:
Adding to @gabikliot note. When developing a storage provider it's important to think of what happens on "duplicate activation". If two duplicate grains attempt a write to the db at the same time, one of them must fail. This is easily achieved by making sure all write operations send the expected version with the data. I think of it as a conditional upsert operation.